### PR TITLE
Update "Sending password by Nextcloud Talk failed" sentence

### LINKED ISSUE #26973: Files Sharing: Unclear meaning of "Sharing sending the password by Nextcloud Talk failed because Nextcloud Talk is not enabled"
Reported at Transifex. See https://www.transifex.com/nextcloud/nextcloud/translate/#de_DE/$/149442095?q=issue%3Aopen

The sentence "Sharing sending the password by Nextcloud Talk failed because Nextcloud Talk is not enabled" is not clear.

Does this mean:
Sharing "Sending the password by Nextcloud Talk" failed because Nextcloud Talk is not enabled?

If so an idea is:

"Sending the password by Nextcloud Talk" for sharing an item failed because Nextcloud Talk is not enabled.

Or:

Sharing: "Sending the password by Nextcloud Talk" failed because Nextcloud Talk is not enabled? (including Apostrophes)

NC version 21.0.1

### PR DESCRIPTION
**Pull request by szaimen**
Fix https://github.com/nextcloud/server/issues/26973
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -1161,7 +1161,7 @@ public function updateShare(
 
 			if ($sendPasswordByTalk === 'true') {
 				if (!$this->appManager->isEnabledForUser('spreed')) {
-					throw new OCSForbiddenException($this->l->t('Sharing sending the password by Nextcloud Talk failed because Nextcloud Talk is not enabled'));
+					throw new OCSForbiddenException($this->l->t('"Sending the password by Nextcloud Talk" for sharing an item failed because Nextcloud Talk is not enabled.'));
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1164]: its a file or folder, not "an item" ?

Comment (szaimen) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1164]: So then `file/folder` instead of `item` ?

Comment (nickvergessen) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1164]: grafik


https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/Share/IShare.php#L221-L228

Make a switch with getNodeType and have 2 translation strings

Comment (szaimen) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1164]: Hm... instructions not clear enough. For getNodeType() we need the path and don't have it here at hand, only the file id, no?

Comment (szaimen) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1164]: I'd vote for just changing the line to file/folder as it is only an error message and not that important...

Comment (nickvergessen) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1164]: `$share->getNodeType()` should work just fine?
But I guess file/folder is also okay

**Review by rakekniven — COMMENTED** [review]
"Sending the password by Nextcloud Talk" for sharing a file or folder failed because Nextcloud Talk is not enabled.

Jm2c.

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 1 above)
Comment (szaimen) [reply apps/files_sharing/lib/Controller/ShareAPIController.php]: I've gone with https://github.com/nextcloud/server/pull/27847#pullrequestreview-700896481

**Review by rakekniven — CHANGES_REQUESTED** [review]
See my suggestion as comment above.

**Comment by szaimen** [comment]
> See my suggestion as comment above.

I did implement it as you suggested...

**Review by rakekniven — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**